### ISSUE TITLE
Nested variant reference broken on 0.6.0-alpha.3

### ISSUE THREAD

**Issue by fcisio** [system-ui/theme-ui#1357]
**Describe the bug**
References of variants in other variants seem to be broken on `0.6.0-alpha.3`

**To Reproduce**
1. Create a new variable to be referenced
2. Use that variant in another variant

```js
variantname: {
    size: {
      size: '100%',
      bg: 'primary'
    },
    round: {
      variant: 'variantname.size',
      overflow: 'hidden',
      borderRadius: '50%',
    }
  }
```
Outputs:

```css
.element {
  variant: variantname.size;
  overflow: hidden;
  border-radius: 50%
}
```

**Expected behavior**
Should output:

```css
.element {
  width: 100%;
  height: 100%;
  background-color: blue;
  overflow: hidden;
  border-radius: 50%
}
```
**Screenshots**

![Shot_0 11](https://user-images.githubusercontent.com/46095852/101954824-c3d6d100-3bca-11eb-8bdd-2498808f88c6.jpg)

**Additional context**
This bug happened on a fresh install of Gatsby, with not many dependencies that could cause conflicts.

**Comment by atanasster** [comment]
Thanks a lot @fcisio for finding this bug. I am adding a PR and tests to catch such issue in the future.

**Comment by atanasster** [comment]
fixed with #1358
